src/pykeen/sampling/negative_sampler.py

```diff
@@ -42,6 +42,10 @@ def __init__(
         self.corruption_scheme = corruption_scheme or ('h', 't')
         # Set the indices
         self._corruption_indices = [0 if side == 'h' else 1 if side == 'r' else 2 for side in self.corruption_scheme]
+        # Copy the mapped triples to the device for efficient filtering
+        if filtered:
+            self.mapped_triples = self.triples_factory.mapped_triples
```

Comment:
This can be deferred to the time when the filter method is called the first time (otherwise there is duplicate code).